File: SW9_Project/MobileGesture.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace SW9_Project {
    class MobileGesture {
        public string Shape;
        public GestureType Type;
        public GestureDirection Direction;
        public DateTime Timestamp;

        public MobileGesture(dynamic jsonObject) {
            Shape = jsonObject["Shape"];
            switch ((string)jsonObject["Type"]) {
                case "Throw": Type = GestureType.Throw; break;
                case "Pinch": Type = GestureType.Pinch; break;
                case "Tilt": Type = GestureType.Tilt; break;
                case "Swipe": Type = GestureType.Swipe; break;
                default: break;
            }
            switch ((string)jsonObject["Direction"]) {
                case "Pull": Direction = GestureDirection.Pull; break;
                case "Push": Direction = GestureDirection.Push; break;
                default: break;
            }
            Timestamp = DateTime.Now;
        }

    }
}

File: Android/app/src/main/java/com/nui/android/MobileGesture.java
package com.nui.android;

/**
 * Created by ericv on 10/13/2015.
 */
public class MobileGesture {

    public MobileGesture(String shape, String type, String direction){
        Type = type;
        Shape = shape;
        Direction = direction;
    }

    public MobileGesture(String shape, String type){
        this(shape, type, "");
    }

    public String Type;
    public String Shape;
    public String Direction;

}
